File: backend/FinanceTracker/FinanceTracker.Domain/Entities/Rule.cs
namespace FinanceTracker.Domain.Entities;

public class Rule
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Field { get; set; } = string.Empty;
    public string Operator { get; set; } = string.Empty;
    public string Value { get; set; } = string.Empty;
    public Guid? CategoryId { get; set; }
    public bool IsEnabled { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public Category? Category { get; set; }
}

File: Frontend/finance-tracker-ui/src/types/rule.ts
export type Rule = {
  id: string;
  name: string;
  field: string;
  operator: string;
  value: string;
  categoryId?: string | null;
  isEnabled: boolean;
};

export type UpsertRuleInput = {
  name: string;
  field: string;
  operator: string;
  value: string;
  categoryId?: string | null;
  isEnabled: boolean;
};
